pkg/registry/populator.go

```diff
@@ -63,7 +63,55 @@ func (i *DirectoryPopulator) Populate(mode Mode) error {
 	return nil
 }
 
+func (i *DirectoryPopulator) globalSanityCheck(imagesToAdd []*ImageInput) error {
+	var errs []error
+	images := make(map[string]bool)
+	for _, image := range imagesToAdd {
+		images[image.bundle.BundleImage] = true
+	}
+
+	for _, image := range imagesToAdd {
```

Comment:
this is making a lot of queries that could be wrapped up into larger sql queries, but it's not worth doing that now, IMO

Reply:
Agreed, we should probably just load the entire package in once, but it doesn't need to happen now.

Reply:
+1